code/modules/atmospherics/environmental/LINDA_system.dm

```diff
@@ -67,18 +67,17 @@
 		var/turf/T = get_step(src, direction)
 		if(!istype(T))
 			continue
-		var/counterdir = get_dir(T, src)
 		if(CanAtmosPass(T))
 			atmos_adjacent_turfs_amount += 1
-			atmos_adjacent_turfs |= direction
-			if(!(T.atmos_adjacent_turfs & counterdir))
+			atmos_adjacent_turfs |= T
+			if(!(src in T.atmos_adjacent_turfs))

```

Comment:
@MrStonedOne 
Real issue here is that trying to add self to the enemy turf will not prevent the enemy turfs from checking this turf in the future AND trying to add the then-sec turf to the then-enemy current one
Basically whole attempt to add self to the enemy turfs list wshould be removed